scipy/stats/tests/test_stats.py

```diff
@@ -4959,7 +4993,10 @@ def test_ttest_rel_axis_size_zero(b, expected_shape):
     # The results should be arrays containing nan with shape
     # given by the broadcast nonaxis dimensions.
     a = np.empty((3, 1, 0))
-    result = stats.ttest_rel(a, b, axis=-1)
+    with np.testing.suppress_warnings() as sup:
+        # first case should warn, second shouldn't?
```

Comment:
Here's what this comment is about:
```python3
import numpy as np
from scipy.stats import skew
skew(np.ones((0, 1)))  # array([nan]) and warning
# SmallSampleWarning: All axis-slices of one or more sample arguments are too small; all elements of returned arrays will be NaN. See documentation for sample size requirements.
skew(np.ones((0, 0)))  # array([], dtype=float64)
```
Both of these calls take the statistic along a slice with length 0. However, the result of the second one is an empty array, so no NaNs actually get produced, and the warning seems unnecessary, so I didn't think the output was surprising enough to deserve a warning. LMK if you disagree.